src/cluster/Backend.h

```diff
@@ -269,6 +269,31 @@ class Backend {
      */
     bool Unsubscribe(const std::string& topic_prefix) { return DoUnsubscribe(topic_prefix); }
 
+    /**
+     * Information passed to a ready callback.
+     */
+    using ReadyCallbackInfo = SubscriptionCallbackInfo;
+
+    using ReadyCallback = std::function<void(const ReadyCallbackInfo& info)>;
+
+    /**
+     * Register a "ready to publish" callback.
+     *
+     * Some cluster backend implementations may not be immediately ready for
+     * publish operations. For example, ZeroMQ has sender-side subscription
+     * filtering. Messages are being discarded when a XPUB socket hasn't
+     * learned about other subscriptions yet.
+     *
+     * This mechanism allows backends to notify the caller that it has now
+     * determined readiness for publish operations.
+     *
+     * Callers should be prepared that \a cb is invoked immediately as that
+     * is the default implementation for DoReadyToPublishCallback().
+     *
+     * @param cb The callback to invoke when the backend is ready for publish operations.
+     */
+    void ReadyToPublishCallback(ReadyCallback cb) { DoReadyToPublishCallback(std::move(cb)); }
```

Comment:
Nit: this could pass `cb` by const ref (and downstream changes in signature of `DoReadyToPublishCallback` and captures of `scb` in `ZeroMQBackend::DoReadyToPublishCallback`).